tests/test_decorator.py

```diff
@@ -68,8 +68,87 @@ def test_cache_response_obj() -> None:
         assert get_cache_response.headers.get("cache-control")
         assert get_cache_response.headers.get("etag")
 
+
 def test_kwargs() -> None:
     with TestClient(app) as client:
         name = "Jon"
-        response = client.get("/kwargs", params = {"name": name})
+        response = client.get("/kwargs", params={"name": name})
         assert response.json() == {"name": name}
+
+
+def test_cache_control_no_store() -> None:
+    with TestClient(app) as client:
+        # Cache-Control: no-store should return a fresh result that is not stored,
+        # so two hits within the cache TTL should be different, and then
+        # removing the removing the header should also not have stored anything.
+
+        # First ensure the cache has expired
+        time.sleep(3)
+
+        response = client.get("/datetime", headers={"cache-control": "no-store"})
+        response_time_1 = response.json().get("now")
+        time.sleep(1)
```

Comment:
You know, one way to fix this (which would not involve making tests sleep, which is nice) would be to test against an API that does something like:

```
import threading
lock = threading.Lock()
counter = 0

@app.get("/counter")
@cache(namespace="test", expire=2)
async def get_counter(request: Request, response: Response):
    with lock:
      counter += 1
    return {"count": counter}
```